www/js/lib/uiUtil.js

```diff
@@ -438,14 +441,21 @@ define(rqDef, function() {
         // is not dark (but we want it applied when the content is dark or inverted)
         footer.classList.add(contentTheme || '_light');
         // Embed a reference to applied theme, so we can remove it generically in the future
-        htmlEl.dataset.theme = theme;
+        htmlEl.dataset.theme = appTheme + contentTheme;
         // Hide any previously displayed help
-        var oldHelp = document.getElementById(oldTheme + '-help');
+        var oldHelp = document.getElementById(oldContentTheme.replace(/_/, '') + '-help');
         if (oldHelp) oldHelp.style.display = 'none';
         // Show any specific help for selected contentTheme
-        var help = document.getElementById(theme + '-help');
+        var help = document.getElementById(contentTheme.replace(/_/, '') + '-help');
         if (help) help.style.display = 'block';
-        
+        // Remove the contentTheme for auto themes whenever system is in light mode
+        if (/^auto/.test(theme) && appTheme === 'light') contentTheme = null;
+        // Hide any previously displayed discription for auto themes
+        var oldDescription = document.getElementById('kiwix-auto-description');
+        oldDescription.style.display = 'none';
```

Comment:
It might be safer to test if oldDescription is defined before using it here (just like elsewhere)

Reply:
>I don't think there is a need to hide the note after a few seconds, but I leave this conversation open in case you want to discuss more about it

@mossroy I was just wanted to testing if this will look okay but later I decided on j=keeping it simple cuz this one was looking better.

>It might be safer to test if oldDescription is defined before using it here (just like elsewhere)

Ohkay, will move this 🚀

Reply:
@Maneet-k Is this minor thing still pending?

Reply:
Oh I'm sorry @Jaifroid , it totally slipped my mind.

Reply:
@Jaifroid as suggested by you that we need to follow the style pattern. Don't you think as `oldHelp` was defined just before using it, So should `oldDescription` follow this pattern. It's just my reason for describing this here. If you want me to fix this, I'm on it.

Reply:
If there is no element in the DOM with name `kiwix-auto-description`, then this line will fail to set a style on it.

Reply:
Oh yess! I got it

Reply:
I don't see how your latest commit fixes the risk that oldDescription could be null. You just moved its instanciation, which does not fix anything.
`getElementById` can return null if the DOM element is not found. See https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById
If this happens, `oldDescription.style.display` will raise an exception.
You can avoid this by testing `if (oldDescription)` before this statement , like it's done for `description`, `help` etc.